Drivers/IWDG_driver/Src/IndependentWatchdog.cpp

```diff
@@ -7,39 +7,54 @@
  *
  */
 IWDG_t::IWDG_t(IWDG_HandleTypeDef *watchdog) {
+	this->watchdog = watchdog;
 	IWDG_InitTypeDef initDef = watchdog->Init;
 	this->iwdg_prescaler = initDef.Prescaler;
 	this->iwdg_reload = initDef.Reload;
 }
 
+
 /**
- * @brief Initalize the driver using custom IWDG_Prescaler and realod values.
+ * @brief Reset the reload and counter values inside a watchdog
+ * using the member variables of IWDG_t
  *
  */
-IWDG_t::IWDG_t(uint8_t iwdg_prescaler, uint16_t iwdg_reload) {
-  // assert(IS_IWDG_PRESCALER(iwdg_prescaler)) Should this be added since it is needed when reinitialzing the watchdog
-  // assert(IS_IWDG_RELOAD(iwdg_reload))
-	this->iwdg_prescaler = iwdg_prescaler;
-	this->iwdg_reload = iwdg_reload;
+bool IWDG_t::SetWatchdogParameters(uint8_t prescaler, uint16_t reload) {
+	this->iwdg_prescaler = prescaler;
+	this->iwdg_reload = reload;
+
+	if (!IS_IWDG_PRESCALER(prescaler) || !IS_IWDG_RELOAD(reload)) return false;
```

Comment:
Where are these macros defined? I didn't see them in my skim of the HAL docs

Reply:
I believe the HAL docs do not mention it even though it is available. I saw that the f4 did not mention it either even though it is available in the stm32f4xx_iwdg.h file. However the function is removed in the newest commit so that should resolve any potential issues.